test/gurka/test_avoids.cc

```diff
@@ -0,0 +1,287 @@
+  return sb.GetString();
+}
+} // namespace
+
+// parameterized test class to test all costings
+class AvoidTest : public ::testing::TestWithParam<std::string> {
+protected:
+  static gurka::map avoid_map;
+
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(A------B
+                                     |      |
+                                     |      |
+                                     D------E)";
+    const gurka::ways ways = {{"AB", {{"highway", "residential"}, {"name", "High"}}},
+                              {"DE", {{"highway", "residential"}, {"name", "Low"}}},
+                              {"AD", {{"highway", "residential"}, {"name", "1st"}}},
+                              {"BE", {{"highway", "residential"}, {"name", "2nd"}}}};
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+    // Add low length limit for avoid_polygons so it throws an error
+    avoid_map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_shortcut",
+                                  {{"service_limits.max_avoid_polygons_length", "1"}});
+  }
+};
+
+gurka::map AvoidTest::avoid_map = {};
+
+TEST_F(AvoidTest, TestLength) {
+  std::vector<vl::line_bg_t> lines(2);
+  // ~ 7.6 km length
+  bg::read_wkt(
+      "LINESTRING (13.38625361 52.4652558, 13.38625361 52.48000128, 13.4181769 52.48000128,13.38625361 52.4652558)",
+      lines[0]);
+  // ~ 10.1 km length
+  bg::read_wkt(
+      "LINESTRING (13.371981 52.439271, 13.371981 52.479088, 13.381839 52.479088, 13.381839 52.439271, 13.371981 52.439271)",
+      lines[1]);
+  double actual_length = bg::length(lines[0], Haversine());
+  actual_length += bg::length(lines[1], Haversine());
+
+  // Add polygons to PBF and calculate total circumference
+  Options options;
+  double calc_length = 0;
+  std::vector<vl::ring_bg_t> e_rings(2);
+  bg::read_wkt(
+      "POLYGON ((13.38625361 52.4652558, 13.38625361 52.48000128, 13.4181769 52.48000128,13.38625361 52.4652558))",
+      e_rings[0]);
+  bg::read_wkt(
+      "POLYGON ((13.371981 52.439271, 13.371981 52.479088, 13.381839 52.479088, 13.381839 52.439271, 13.371981 52.439271))",
+      e_rings[1]);
+  auto avoid_polygons = options.mutable_avoid_polygons();
+  for (const auto& ring : e_rings) {
+    auto* polygon = avoid_polygons->Add();
+    for (auto& coord : ring) {
+      auto* ll = polygon->add_coords();
+      ll->set_lng(coord.lng());
+      ll->set_lat(coord.lat());
+    }
+    calc_length += vl::GetRingLength(vl::PBFToRing(*polygon));
+  }
+
+  EXPECT_EQ(actual_length, calc_length);
+}
+
+TEST_F(AvoidTest, TestConfig) {
+  // Add a polygon with longer perimeter than the limit
+  std::vector<vl::ring_bg_t> rings{{{13.38625361, 52.4652558},
+                                    {13.38625361, 52.48000128},
+                                    {13.4181769, 52.48000128},
+                                    {13.4181769, 52.4652558}}};
+
+  auto lls = gurka::detail::to_lls(avoid_map.nodes, {"A", "D"});
```

Comment:
i'm pretty sure the compiler will deduce a `std::vector` type here without you supplying it but if it complains just add that to the left of the first `{`

```suggestion
  auto lls = {avoid_map.nodes["A"], avoid_map.nodes["D"]};
```

Reply:
worked